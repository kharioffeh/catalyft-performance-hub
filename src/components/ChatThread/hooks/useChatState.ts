
import { useState, useEffect } from "react";
import { useLocation, useParams } from "react-router-dom";
import { ariaStream, getAriaMessages, type AriaChatMessage, type AriaMessage } from "@/lib/aria";

type Msg = { id: string; role: "assistant" | "user"; text: string; isStreaming?: boolean };

export function useChatState() {
  const { threadId } = useParams<{ threadId?: string }>();
  const location = useLocation();
  const locationState = location.state as { initialQuestion?: string } | null;
  
  const [messages, setMessages] = useState<Msg[]>([]);
  const [draft, setDraft] = useState("");
  const [isLoading, setIsLoading] = useState(false);
  const [actualThreadId, setActualThreadId] = useState<string | undefined>(threadId);

  // Load existing messages when thread ID changes
  useEffect(() => {
    async function loadThreadMessages() {
      if (threadId) {
        try {
          const ariaMessages = await getAriaMessages(threadId);
          const formattedMessages: Msg[] = ariaMessages
            .filter((msg: AriaMessage) => msg.role !== 'system')
            .map((msg: AriaMessage) => ({
              id: msg.id.toString(),
              role: msg.role as "assistant" | "user",
              text: msg.content
            }));
          setMessages(formattedMessages);
          setActualThreadId(threadId);
        } catch (error) {
          console.error('Error loading thread messages:', error);
          // If we can't load the thread, start fresh
          setMessages([
            { id: "welcome", role: "assistant", text: "Hi ðŸ‘‹ â€” what can I do for you?" },
          ]);
        }
      } else {
        // No thread ID - start with welcome message
        setMessages([
          { id: "welcome", role: "assistant", text: "Hi ðŸ‘‹ â€” what can I do for you?" },
        ]);
      }
    }

    loadThreadMessages();
  }, [threadId]);

  // Handle initial question from navigation state
  useEffect(() => {
    if (locationState?.initialQuestion && !threadId) {
      // Start a new conversation with the initial question
      sendMessage(locationState.initialQuestion);
    }
  }, [locationState?.initialQuestion, threadId]);

  async function sendMessage(messageText?: string) {
    const messageToSend = messageText || draft.trim();
    if (!messageToSend || isLoading) return;

    const userMsg: Msg = { 
      id: crypto.randomUUID(), 
      role: "user", 
      text: messageToSend 
    };
    
    setMessages((m) => [...m, userMsg]);
    if (!messageText) setDraft("");
    setIsLoading(true);

    // Add streaming assistant message placeholder
    const assistantMsgId = crypto.randomUUID();
    const assistantMsg: Msg = {
      id: assistantMsgId,
      role: "assistant",
      text: "",
      isStreaming: true
    };
    setMessages((m) => [...m, assistantMsg]);

    try {
      // Prepare messages for ARIA
      const conversationHistory: AriaChatMessage[] = [
        ...messages.map(msg => ({
          role: msg.role,
          content: msg.text
        })),
        {
          role: 'user' as const,
          content: messageToSend
        }
      ];

      // Call ARIA with streaming
      const response = await ariaStream(
        conversationHistory, 
        actualThreadId,
        (token: string) => {
          // Update the streaming message with each token
          setMessages((currentMessages) => 
            currentMessages.map(msg => 
              msg.id === assistantMsgId 
                ? { ...msg, text: msg.text + token }
                : msg
            )
          );
        }
      );

      // Update thread ID if this was a new conversation
      if (response.thread_id && !actualThreadId) {
        setActualThreadId(response.thread_id);
        // Update the URL without navigation to reflect the new thread
        window.history.replaceState({}, '', `/chat/${response.thread_id}`);
      }

      // Mark streaming as complete
      setMessages((currentMessages) => 
        currentMessages.map(msg => 
          msg.id === assistantMsgId 
            ? { ...msg, isStreaming: false }
            : msg
        )
      );

    } catch (error) {
      console.error('Error sending message:', error);
      
      // Replace streaming message with error
      setMessages((currentMessages) => 
        currentMessages.map(msg => 
          msg.id === assistantMsgId 
            ? { 
                ...msg, 
                text: "I'm sorry, I encountered an error. Please try again.",
                isStreaming: false
              }
            : msg
        )
      );
    } finally {
      setIsLoading(false);
    }
  }

  return {
    messages,
    draft,
    setDraft,
    isLoading,
    actualThreadId,
    sendMessage
  };
}
