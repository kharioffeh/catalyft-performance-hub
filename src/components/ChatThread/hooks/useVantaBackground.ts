
import { useEffect, useRef } from "react";
import * as THREE from "three";
import NET from "vanta/dist/vanta.net.min";

export function useVantaBackground(ref: React.RefObject<HTMLDivElement>) {
  const vantaRef = useRef<any>(null);
  
  useEffect(() => {
    if (!ref.current) return;
    
    vantaRef.current = NET({
      el: ref.current,
      THREE,
      mouseControls: true,
      touchControls: true,
      backgroundColor: 0x101014,
      color: 0x5e6ad2,
      points: 8,
      maxDistance: 25,
      spacing: 20,
    });
    
    return () => vantaRef.current?.destroy?.();
  }, [ref]);
}
