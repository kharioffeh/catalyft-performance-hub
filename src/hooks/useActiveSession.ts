import { useQuery } from '@tanstack/react-query';
import { supabase } from '@/integrations/supabase/client';
import { useAuth } from '@/contexts/AuthContext';

interface ActiveSession {
  id: string;
  type: string;
  start_ts: string;
  status: 'active';
  athletes?: {
    name: string;
  };
}

export const useActiveSession = () => {
  const { profile } = useAuth();
  
  return useQuery({
    queryKey: ['activeSession', profile?.id],
    queryFn: async (): Promise<ActiveSession | null> => {
      if (!profile?.id) return null;
      
      const { data, error } = await supabase
        .from('sessions')
        .select(`
          id,
          type,
          start_ts,
          status,
          athletes!inner(name)
        `)
        .eq('status', 'active')
        .eq('athlete_uuid', profile.id)
        .single();
      
      if (error || !data) return null;
      return data as ActiveSession;
    },
    enabled: !!profile?.id,
    refetchInterval: 5000, // Refetch every 5 seconds to stay synced
  });
};