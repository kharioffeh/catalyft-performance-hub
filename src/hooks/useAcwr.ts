
import { useQuery } from '@tanstack/react-query';
import { supabase } from '@/integrations/supabase/client';
import { useAuth } from '@/contexts/AuthContext';

export const useAcwr = () => {
  const { profile } = useAuth();

  return useQuery({
    queryKey: ['acwr', profile?.id],
    queryFn: async () => {
      if (!profile?.id) return null;

      const { data, error } = await supabase
        .from('vw_load_acwr')
        .select('acwr_7_28')
        .eq('athlete_uuid', profile.id)
        .order('day', { ascending: false })
        .limit(1)
        .maybeSingle();

      if (error) {
        console.error('Error fetching ACWR:', error);
        return null;
      }

      return data?.acwr_7_28 || null;
    },
    enabled: !!profile?.id
  });
};
