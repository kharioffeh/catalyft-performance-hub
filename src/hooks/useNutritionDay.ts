import { useCallback } from 'react';
import { useNutrition } from './useNutrition';

export interface NutritionDay {
  kcal: number;
  protein: number;
  carb: number;
  fat: number;
}

/**
 * Hook to get today's nutrition intake
 */
export const useNutritionDay = (): NutritionDay => {
  const { getTodaysMacros } = useNutrition();

  const getTodaysNutrition = useCallback((): NutritionDay => {
    const macros = getTodaysMacros();
    return {
      kcal: macros.calories,
      protein: macros.protein,
      carb: macros.carbs,
      fat: macros.fat,
    };
  }, [getTodaysMacros]);

  return getTodaysNutrition();
};