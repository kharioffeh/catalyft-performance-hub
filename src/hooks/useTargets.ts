import { useCallback } from 'react';
import { useAuth } from '@/contexts/AuthContext';

export interface NutritionTargets {
  kcalTarget: number;
  proteinTarget: number;
  carbTarget: number;
  fatTarget: number;
}

/**
 * Hook to get nutrition targets for the current user
 */
export const useTargets = (): NutritionTargets => {
  const { profile } = useAuth();

  const getTargets = useCallback((): NutritionTargets => {
    // Default targets - could be user-customizable later from profile/program
    return {
      kcalTarget: 2200,
      proteinTarget: 120,
      carbTarget: 250,
      fatTarget: 80,
    };
  }, [profile]);

  return getTargets();
};