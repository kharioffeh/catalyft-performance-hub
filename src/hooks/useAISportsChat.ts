
import { useState } from 'react';
import { supabase } from '@/integrations/supabase/client';
import { useAuth } from '@/contexts/AuthContext';
import { toast } from '@/hooks/use-toast';

interface ChatMessage {
  id: string;
  role: 'user' | 'assistant';
  content: string;
  timestamp: Date;
}

export const useAISportsChat = () => {
  const [messages, setMessages] = useState<ChatMessage[]>([]);
  const [isLoading, setIsLoading] = useState(false);
  const { profile } = useAuth();

  const sendMessage = async (message: string) => {
    if (!profile?.id) {
      toast({
        title: "Authentication Required",
        description: "Please log in to use the AI chat assistant",
        variant: "destructive",
      });
      return;
    }

    const userMessage: ChatMessage = {
      id: Date.now().toString(),
      role: 'user',
      content: message,
      timestamp: new Date(),
    };

    setMessages(prev => [...prev, userMessage]);
    setIsLoading(true);

    try {
      const { data, error } = await supabase.functions.invoke('ai-sports-chat', {
        body: {
          message,
          athleteId: profile.id,
        },
      });

      if (error) throw error;

      const assistantMessage: ChatMessage = {
        id: (Date.now() + 1).toString(),
        role: 'assistant',
        content: data.response,
        timestamp: new Date(),
      };

      setMessages(prev => [...prev, assistantMessage]);

    } catch (error) {
      console.error('Chat error:', error);
      toast({
        title: "Chat Error",
        description: "Failed to get AI response. Please try again.",
        variant: "destructive",
      });
    } finally {
      setIsLoading(false);
    }
  };

  const clearChat = () => {
    setMessages([]);
  };

  return {
    messages,
    isLoading,
    sendMessage,
    clearChat,
  };
};
