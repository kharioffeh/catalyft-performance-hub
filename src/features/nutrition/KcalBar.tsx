import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import Animated, {
  useSharedValue,
  useAnimatedStyle,
  withSpring,
  interpolateColor,
} from 'react-native-reanimated';
import { useTargets } from '@/hooks/useTargets';
import { useNutritionDay } from '@/hooks/useNutritionDay';

export const KcalBar: React.FC = () => {
  const { kcalTarget } = useTargets();
  const { kcal } = useNutritionDay();

  // Calculate progress percentage
  const progressPercentage = kcalTarget > 0 ? (kcal / kcalTarget) * 100 : 0;
  const progressValue = useSharedValue(0);

  // Update animated value when data changes
  React.useEffect(() => {
    progressValue.value = withSpring(Math.min(progressPercentage, 100), {
      damping: 15,
      stiffness: 150,
    });
  }, [progressPercentage, progressValue]);

  // Animated styles for the progress bar
  const animatedBarStyle = useAnimatedStyle(() => {
    const backgroundColor = interpolateColor(
      progressValue.value,
      [0, 100, 150],
      ['hsl(var(--primary))', 'hsl(var(--primary))', '#f97316'] // blue to orange
    );

    return {
      width: `${progressValue.value}%`,
      backgroundColor,
    };
  });

  // Animated styles for overflow bar (when >100%)
  const overflowBarStyle = useAnimatedStyle(() => {
    const overflowPercentage = Math.max(0, progressPercentage - 100);
    const width = Math.min(overflowPercentage, 50); // Cap at 50% additional width
    
    return {
      width: `${width}%`,
      backgroundColor: '#f97316', // orange for overflow
      opacity: progressPercentage > 100 ? 1 : 0,
    };
  });

  const formattedPercentage = progressPercentage.toFixed(1);

  return (
    <View style={styles.container}>
      {/* Header */}
      <View style={styles.header}>
        <Text style={styles.title}>Calories</Text>
        <Text style={styles.values}>
          {Math.round(kcal)} / {Math.round(kcalTarget)} kcal
        </Text>
      </View>

      {/* Progress Bar Container */}
      <View style={styles.progressContainer}>
        <View style={styles.progressTrack}>
          {/* Main Progress Bar */}
          <Animated.View style={[styles.progressBar, animatedBarStyle]} />
          
          {/* Overflow Bar (when >100%) */}
          {progressPercentage > 100 && (
            <Animated.View style={[styles.overflowBar, overflowBarStyle]} />
          )}
        </View>
      </View>

      {/* Percentage Display */}
      <View style={styles.footer}>
        <Text 
          style={[
            styles.percentage,
            progressPercentage > 100 && styles.overPercentage
          ]}
        >
          {formattedPercentage}%
        </Text>
        {progressPercentage > 100 && (
          <Text style={styles.overTarget}>Over target</Text>
        )}
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: 16,
    backgroundColor: 'hsl(var(--card))',
    borderRadius: 12,
    borderWidth: 1,
    borderColor: 'hsl(var(--border))',
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: 12,
  },
  title: {
    fontSize: 16,
    fontWeight: '600',
    color: 'hsl(var(--foreground))',
  },
  values: {
    fontSize: 14,
    color: 'hsl(var(--muted-foreground))',
  },
  progressContainer: {
    marginBottom: 8,
  },
  progressTrack: {
    height: 8,
    backgroundColor: 'hsl(var(--muted))',
    borderRadius: 4,
    overflow: 'hidden',
    position: 'relative',
  },
  progressBar: {
    height: '100%',
    borderRadius: 4,
    position: 'absolute',
    left: 0,
    top: 0,
  },
  overflowBar: {
    height: '100%',
    borderRadius: 4,
    position: 'absolute',
    left: '100%',
    top: 0,
  },
  footer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  percentage: {
    fontSize: 14,
    fontWeight: '500',
    color: 'hsl(var(--foreground))',
  },
  overPercentage: {
    color: '#f97316', // orange when over target
  },
  overTarget: {
    fontSize: 12,
    color: '#f97316',
    fontWeight: '500',
  },
});