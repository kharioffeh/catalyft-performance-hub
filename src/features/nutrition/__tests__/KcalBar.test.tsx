import React from 'react';
import { render } from '@testing-library/react';
import { KcalBar } from '../KcalBar';
import { useTargets } from '@/hooks/useTargets';
import { useNutritionDay } from '@/hooks/useNutritionDay';

// Mock the hooks
jest.mock('@/hooks/useTargets');
jest.mock('@/hooks/useNutritionDay');

const mockUseTargets = useTargets as jest.MockedFunction<typeof useTargets>;
const mockUseNutritionDay = useNutritionDay as jest.MockedFunction<typeof useNutritionDay>;

// Mock react-native-reanimated
jest.mock('react-native-reanimated', () => {
  const Reanimated = require('react-native-reanimated/mock');
  
  // Add missing mocks
  Reanimated.default.View = Reanimated.View;
  Reanimated.interpolateColor = jest.fn((value, inputRange, outputRange) => outputRange[0]);
  
  return Reanimated;
});

describe('KcalBar Snapshots', () => {
  const mockTargets = {
    kcalTarget: 2200,
    proteinTarget: 120,
    carbTarget: 250,
    fatTarget: 80,
  };

  beforeEach(() => {
    jest.clearAllMocks();
    mockUseTargets.mockReturnValue(mockTargets);
  });

  it('renders correctly at 0% progress', () => {
    mockUseNutritionDay.mockReturnValue({
      kcal: 0,
      protein: 0,
      carb: 0,
      fat: 0,
    });

    const tree = render(<KcalBar />);
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly at 100% progress', () => {
    mockUseNutritionDay.mockReturnValue({
      kcal: 2200,
      protein: 120,
      carb: 250,
      fat: 80,
    });

    const tree = render(<KcalBar />);
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly at 150% progress (over target)', () => {
    mockUseNutritionDay.mockReturnValue({
      kcal: 3300,
      protein: 180,
      carb: 375,
      fat: 120,
    });

    const tree = render(<KcalBar />);
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly at 50% progress', () => {
    mockUseNutritionDay.mockReturnValue({
      kcal: 1100,
      protein: 60,
      carb: 125,
      fat: 40,
    });

    const tree = render(<KcalBar />);
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly at 120% progress (slightly over)', () => {
    mockUseNutritionDay.mockReturnValue({
      kcal: 2640,
      protein: 144,
      carb: 300,
      fat: 96,
    });

    const tree = render(<KcalBar />);
    expect(tree).toMatchSnapshot();
  });
});

describe('KcalBar Functionality', () => {
  const mockTargets = {
    kcalTarget: 2000,
    proteinTarget: 100,
    carbTarget: 200,
    fatTarget: 70,
  };

  beforeEach(() => {
    jest.clearAllMocks();
    mockUseTargets.mockReturnValue(mockTargets);
  });

  it('calculates percentage correctly for various intake levels', () => {
    // Test 0%
    mockUseNutritionDay.mockReturnValue({
      kcal: 0,
      protein: 0,
      carb: 0,
      fat: 0,
    });

    const { container } = render(<KcalBar />);
    expect(container.textContent).toContain('0 / 2000 kcal');
    expect(container.textContent).toContain('0.0%');
  });

  it('shows over target message when exceeding 100%', () => {
    mockUseNutritionDay.mockReturnValue({
      kcal: 2500,
      protein: 125,
      carb: 250,
      fat: 87,
    });

    const { container } = render(<KcalBar />);
    expect(container.textContent).toContain('2500 / 2000 kcal');
    expect(container.textContent).toContain('125.0%');
    expect(container.textContent).toContain('Over target');
  });

  it('rounds percentage to one decimal place', () => {
    mockUseNutritionDay.mockReturnValue({
      kcal: 1333, // This should give 66.65% which rounds to 66.7%
      protein: 67,
      carb: 133,
      fat: 47,
    });

    const { container } = render(<KcalBar />);
    expect(container.textContent).toContain('66.7%');
  });

  it('handles zero target gracefully', () => {
    mockUseTargets.mockReturnValue({
      kcalTarget: 0,
      proteinTarget: 0,
      carbTarget: 0,
      fatTarget: 0,
    });

    mockUseNutritionDay.mockReturnValue({
      kcal: 1000,
      protein: 50,
      carb: 100,
      fat: 35,
    });

    const { container } = render(<KcalBar />);
    expect(container.textContent).toContain('1000 / 0 kcal');
    expect(container.textContent).toContain('0.0%'); // Should default to 0% when target is 0
  });

  it('reacts to changes in both hooks', () => {
    const { rerender } = render(<KcalBar />);

    // Initial state
    mockUseTargets.mockReturnValue(mockTargets);
    mockUseNutritionDay.mockReturnValue({ kcal: 1000, protein: 50, carb: 100, fat: 35 });

    rerender(<KcalBar />);
    expect(document.body.textContent).toContain('50.0%');

    // Change nutrition day
    mockUseNutritionDay.mockReturnValue({ kcal: 1500, protein: 75, carb: 150, fat: 52 });

    rerender(<KcalBar />);
    expect(document.body.textContent).toContain('75.0%');

    // Change targets
    mockUseTargets.mockReturnValue({
      kcalTarget: 3000,
      proteinTarget: 150,
      carbTarget: 300,
      fatTarget: 105,
    });

    rerender(<KcalBar />);
    expect(document.body.textContent).toContain('50.0%'); // 1500/3000 = 50%
  });
});