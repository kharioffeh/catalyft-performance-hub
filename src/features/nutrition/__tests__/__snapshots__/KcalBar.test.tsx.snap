// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KcalBar Snapshots renders correctly at 0% progress 1`] = `
{
  "container": <div>
    <div
      style="padding: 16px; background-color: hsl(var(--card)); border-radius: 12px; border-width: 1px; border-color: hsl(var(--border));"
    >
      <div
        style="flex-direction: row; justify-content: space-between; align-items: center; margin-bottom: 12px;"
      >
        <div
          style="font-size: 16px; font-weight: 600; color: hsl(var(--foreground));"
        >
          Calories
        </div>
        <div
          style="font-size: 14px; color: hsl(var(--muted-foreground));"
        >
          0 / 2200 kcal
        </div>
      </div>
      <div
        style="margin-bottom: 8px;"
      >
        <div
          style="height: 8px; background-color: hsl(var(--muted)); border-radius: 4px; overflow: hidden; position: relative;"
        >
          <div
            style="height: 100%; border-radius: 4px; position: absolute; left: 0px; top: 0px; width: 0%;"
          />
        </div>
      </div>
      <div
        style="flex-direction: row; justify-content: space-between; align-items: center;"
      >
        <div
          style="font-size: 14px; font-weight: 500; color: hsl(var(--foreground));"
        >
          0.0%
        </div>
      </div>
    </div>
  </div>,
}
`;

exports[`KcalBar Snapshots renders correctly at 100% progress 1`] = `
{
  "container": <div>
    <div
      style="padding: 16px; background-color: hsl(var(--card)); border-radius: 12px; border-width: 1px; border-color: hsl(var(--border));"
    >
      <div
        style="flex-direction: row; justify-content: space-between; align-items: center; margin-bottom: 12px;"
      >
        <div
          style="font-size: 16px; font-weight: 600; color: hsl(var(--foreground));"
        >
          Calories
        </div>
        <div
          style="font-size: 14px; color: hsl(var(--muted-foreground));"
        >
          2200 / 2200 kcal
        </div>
      </div>
      <div
        style="margin-bottom: 8px;"
      >
        <div
          style="height: 8px; background-color: hsl(var(--muted)); border-radius: 4px; overflow: hidden; position: relative;"
        >
          <div
            style="height: 100%; border-radius: 4px; position: absolute; left: 0px; top: 0px; width: 100%;"
          />
        </div>
      </div>
      <div
        style="flex-direction: row; justify-content: space-between; align-items: center;"
      >
        <div
          style="font-size: 14px; font-weight: 500; color: hsl(var(--foreground));"
        >
          100.0%
        </div>
      </div>
    </div>
  </div>,
}
`;

exports[`KcalBar Snapshots renders correctly at 150% progress (over target) 1`] = `
{
  "container": <div>
    <div
      style="padding: 16px; background-color: hsl(var(--card)); border-radius: 12px; border-width: 1px; border-color: hsl(var(--border));"
    >
      <div
        style="flex-direction: row; justify-content: space-between; align-items: center; margin-bottom: 12px;"
      >
        <div
          style="font-size: 16px; font-weight: 600; color: hsl(var(--foreground));"
        >
          Calories
        </div>
        <div
          style="font-size: 14px; color: hsl(var(--muted-foreground));"
        >
          3300 / 2200 kcal
        </div>
      </div>
      <div
        style="margin-bottom: 8px;"
      >
        <div
          style="height: 8px; background-color: hsl(var(--muted)); border-radius: 4px; overflow: hidden; position: relative;"
        >
          <div
            style="height: 100%; border-radius: 4px; position: absolute; left: 0px; top: 0px; width: 100%;"
          />
          <div
            style="height: 100%; border-radius: 4px; position: absolute; left: 100%; top: 0px; width: 50%; background-color: rgb(249, 115, 22); opacity: 1;"
          />
        </div>
      </div>
      <div
        style="flex-direction: row; justify-content: space-between; align-items: center;"
      >
        <div
          style="font-size: 14px; font-weight: 500; color: rgb(249, 115, 22);"
        >
          150.0%
        </div>
        <div
          style="font-size: 12px; color: rgb(249, 115, 22); font-weight: 500;"
        >
          Over target
        </div>
      </div>
    </div>
  </div>,
}
`;

exports[`KcalBar Snapshots renders correctly at 50% progress 1`] = `
{
  "container": <div>
    <div
      style="padding: 16px; background-color: hsl(var(--card)); border-radius: 12px; border-width: 1px; border-color: hsl(var(--border));"
    >
      <div
        style="flex-direction: row; justify-content: space-between; align-items: center; margin-bottom: 12px;"
      >
        <div
          style="font-size: 16px; font-weight: 600; color: hsl(var(--foreground));"
        >
          Calories
        </div>
        <div
          style="font-size: 14px; color: hsl(var(--muted-foreground));"
        >
          1100 / 2200 kcal
        </div>
      </div>
      <div
        style="margin-bottom: 8px;"
      >
        <div
          style="height: 8px; background-color: hsl(var(--muted)); border-radius: 4px; overflow: hidden; position: relative;"
        >
          <div
            style="height: 100%; border-radius: 4px; position: absolute; left: 0px; top: 0px; width: 50%;"
          />
        </div>
      </div>
      <div
        style="flex-direction: row; justify-content: space-between; align-items: center;"
      >
        <div
          style="font-size: 14px; font-weight: 500; color: hsl(var(--foreground));"
        >
          50.0%
        </div>
      </div>
    </div>
  </div>,
}
`;

exports[`KcalBar Snapshots renders correctly at 120% progress (slightly over) 1`] = `
{
  "container": <div>
    <div
      style="padding: 16px; background-color: hsl(var(--card)); border-radius: 12px; border-width: 1px; border-color: hsl(var(--border));"
    >
      <div
        style="flex-direction: row; justify-content: space-between; align-items: center; margin-bottom: 12px;"
      >
        <div
          style="font-size: 16px; font-weight: 600; color: hsl(var(--foreground));"
        >
          Calories
        </div>
        <div
          style="font-size: 14px; color: hsl(var(--muted-foreground));"
        >
          2640 / 2200 kcal
        </div>
      </div>
      <div
        style="margin-bottom: 8px;"
      >
        <div
          style="height: 8px; background-color: hsl(var(--muted)); border-radius: 4px; overflow: hidden; position: relative;"
        >
          <div
            style="height: 100%; border-radius: 4px; position: absolute; left: 0px; top: 0px; width: 100%;"
          />
          <div
            style="height: 100%; border-radius: 4px; position: absolute; left: 100%; top: 0px; width: 20%; background-color: rgb(249, 115, 22); opacity: 1;"
          />
        </div>
      </div>
      <div
        style="flex-direction: row; justify-content: space-between; align-items: center;"
      >
        <div
          style="font-size: 14px; font-weight: 500; color: rgb(249, 115, 22);"
        >
          120.0%
        </div>
        <div
          style="font-size: 12px; color: rgb(249, 115, 22); font-weight: 500;"
        >
          Over target
        </div>
      </div>
    </div>
  </div>,
}
`;